File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/Ongoing.java
package com.amakakeru.mangaworld.Entity;

import com.amakakeru.mangaworld.Entity.IdClasses.OngoingId;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@IdClass(OngoingId.class)
@Table(name = "ongoing")
public class Ongoing {

    @Id
    @Column(name = "user_id")
    private Long userId;

    @Id
    @Column(name = "m_id")
    private Long mid;

    @Column(name = "current_volume", nullable = false)
    private int currentVolume;

    @Column(name = "current_chapter", nullable = false)
    private int currentChapter;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;

    @ManyToOne
    @JoinColumn(name = "current_volume", insertable = false, updatable = false)
    private Volume volume;

    @ManyToOne
    @JoinColumn(name = "current_chapter", insertable = false, updatable = false)
    private Chapter chapter;
}

File: Frontend/manga-world/src/Model/Ongoing.ts
import Chapter from "./Chapter";
import Manga from "./Manga";
import User from "./user";
import Volume from "./Volume";

class ongoing {
    userId: number;
    mId: number;
    currentVolume: number;
    currentChapter: number;
    user: User;
    manga: Manga;
    volume: Volume;
    chapter: Chapter;

    constructor(userId: number, mId: number, currentVolume: number, currentChapter: number, user: User, manga: Manga, volume: Volume, chapter: Chapter) {
        this.userId = userId;
        this.mId = mId;
        this.currentVolume = currentVolume;
        this.currentChapter = currentChapter;
        this.user = user;
        this.manga = manga;
        this.volume = volume;
        this.chapter = chapter;
    }
}

export default ongoing;